os/kernel/semaphore/sem_post.c

```diff
@@ -140,9 +143,9 @@ int sem_post(FAR sem_t *sem)
 		saved_state = irqsave();
 
 		/* Perform the semaphore unlock operation. */
-
+		htcb = this_task();
```

Comment:
I think we can use stcb in here.
And if we use htcb, we don't need to use rtcb when CONFIG_SEMAPHORE_HISTORY is enabled.